### ISSUE TITLE
[MacCatalyst] Tab does not keyboard navigate away from Editor control

### ISSUE THREAD

**Issue by rachelkang** [dotnet/maui#14191]
### Description

The keyboard Tab key is used to navigate between interactive controls on most platforms, including Mac.

On Mac, pressing Tab will successfully navigate keyboard focus to and away from Entry controls. It will successfully navigate keyboard focus to Editor controls as well, but not AWAY from them. When pressing Tab key from an Editor field, tabbed spaces will be created within the editor (as if within a docx), rather than navigating keyboard focus to the next interactive control. From an editor, there seems to be no way to navigate away from it using just the keyboard. Is this a keyboard trap or am I missing something?

### Steps to Reproduce

1. Download https://github.com/rachelkang/recipeSearch
2. Deploy to MacCatalyst device
3. Click on "My Recipes" tab
4. Click on "New" button in top right corner
5. Click on keyboard Tab key to navigate through the various entrys and editors
6. Notice how when reaching the first editor, there's no way to navigate away. Tab will just create literal tabbed spaces in the field, and Shift+Tab does nothing

### Link to public reproduction project repository

https://github.com/rachelkang/recipeSearch

### Version with bug

7.0 (current)

### Last version that worked well

7.0 (current)

### Affected platforms

macOS

### Affected platform versions

macOS Ventura 13.2.1

**Comment by Zhanglirong-Winnie** [comment]
Verified this issue with Visual Studio Enterprise 17.6 Preview(17.6.0.1539). Can repro on macOS platform with sample project.
[recipeSearch-main.zip](https://github.com/dotnet/maui/files/11755236/recipeSearch-main.zip)
![14191](https://github.com/dotnet/maui/assets/77434865/637e99be-d579-4dfc-8c74-a30811e40d28)

**Comment by agingo** [comment]
We need this problem fixed. The same problem exists on .NET Core 6 & 7. 

Any updates?
